sklearn/utils/estimator_checks.py

```diff
@@ -276,6 +276,8 @@ def _yield_all_checks(name, estimator):
     yield check_dict_unchanged
     yield check_dont_overwrite_parameters
     yield check_fit_idempotent
+    if not tags["no_validation"]:
```

Comment:
not sure about the tag name. maybe `no_input_validation`?

also, this name refers to what causes the API to exist, instead of referring to the API itself, should the name refer to the API instead? like `no_feature_count_api` or something?

Reply:
The tag already exists, it's not defined in this PR

Reply:
I am the one to blame here. We could maybe update this with a new tag if you think this is problematic

Reply:
It'd be nice to have an open issue around it, but I wouldn't say it blocks this PR.